### DISCUSSION TITLE
Something is wrong about blocks

### DISCUSSION THREAD

**Discussion by bablilayoub** [GrapesJS/grapesjs#5224] [Q&A]
hi i found a problem in this case , so when i add block like that for example :
`editor.BlockManager.add('text', {
                label: 'Text',
                content: {
                    type: 'text',
                    content: 'Insert your text here',
                    activeOnRender: 1,
                    tagName: 'p',
                }
});`
The code works just find and when i export the template the element is exported like this : 
`<p id="izk2b">Insert your text here</p>`
to now everything is fine , but when i add block on this way : 
`editor.BlockManager.add('text', {
                label: 'Text',
                content: {
                    type: 'text',
                    content: 'Insert your text here',
                    activeOnRender: 1,
                    tagName: 'p',
                    classes: ['text-base'],
                }
});`

the results of export template of the element returns this :
`<p class="text-base">Insert your text here</p>`

in this case there is no id which causes a problem when changing styles of element after duplicating the element , because css is being applied on the class not id which means that the both element are going to change when changing styles of one of them , so the problem in short , i just need the id to applied when add class to block

**Comment by thigh** [comment]
Hi @bablilayoub , not sure if this helps but I apply classes to my custom components via the attributes property.
For example,
`attributes: { class: "my-custom-class"}`
when the component is added, all the classes I defined in the class attribute are properly added and the id's are still retained.
